File: src/ios/XRRemoteLog.m
//
//  XRRemoteLog.m
//  cordova_demo
//
//  Created by 袁训锐 on 2020/2/28.
//

#import "XRRemoteLog.h"
// 获取 远程日志
#import <TRemoteDebugger/TLogBiz.h>
#import <TRemoteDebugger/TLogFactory.h>

@implementation XRRemoteLog

- (instancetype)init{
    self = [super init];
    return self;
}
- (TLogBiz *)getLogFactoryWithModuleName:(NSString *)name{
    TLogBiz *log = [TLogFactory createTLogForModuleName:name];
    return log;
}
- (void)error:(CDVInvokedUrlCommand *)command{
    NSString *title = [command.arguments objectAtIndex:0];
    NSString *message = [command.arguments objectAtIndex:1];
    TLogBiz *log = [self getLogFactoryWithModuleName:title];
    [log error:message];
}
- (void)warn:(CDVInvokedUrlCommand *)command{
    NSString *title = [command.arguments objectAtIndex:0];
    NSString *message = [command.arguments objectAtIndex:1];
    TLogBiz *log = [self getLogFactoryWithModuleName:title];
    [log error:message];
}
- (void)debug:(CDVInvokedUrlCommand *)command{
    NSString *title = [command.arguments objectAtIndex:0];
    NSString *message = [command.arguments objectAtIndex:1];
    TLogBiz *log = [self getLogFactoryWithModuleName:title];
    [log error:message];
}
- (void)info:(CDVInvokedUrlCommand *)command{
    NSString *title = [command.arguments objectAtIndex:0];
    NSString *message = [command.arguments objectAtIndex:1];
    TLogBiz *log = [self getLogFactoryWithModuleName:title];
    [log error:message];
}
@end

File: www/XRRemoteLog.js
var exec = require('cordova/exec');

var XRLog = {

    error: function (title, msg) {
        if (!title || !msg || title.length === 0 || msg.length === 0) {
            console.log('参数错误');
            return;
        }
        const arr = [title,msg];
        exec(null, null, 'XRRemoteLog', 'error', arr);
    },

    warn: function (title, msg) {
        if (!title || !msg || title.length === 0 || msg.length === 0) {
            console.log('参数错误');
            return;
        }
        const arr = [title,msg];
        exec(null, null, 'XRRemoteLog', 'warn', arr);
    },
    debug: function (title, msg) {
        if (!title || !msg || title.length === 0 || msg.length === 0) {
            console.log('参数错误');
            return;
        }
        const arr = [title,msg];
        exec(null, null, 'XRRemoteLog', 'debug', arr);
    },
    info: function (title, msg) {
        if (!title || !msg || title.length === 0 || msg.length === 0) {
            console.log('参数错误');
            return;
        }
        const arr = [title,msg];
        exec(null, null, 'XRRemoteLog', 'info', arr);
    }
};

module.exports = XRLog;
